### DISCUSSION TITLE
found weird question.

### DISCUSSION THREAD

**Discussion by magic-thomas** [filamentphp/filament#9315] [Help]
I have laravel 10 project which use filament/filament package 2. 

I cloned and renamed the project. 

I removed livewire and filament package ,

And installed filament package  3 again . 

Yes, I didn't use upgrade package. 

And run `php artisan filament:install --panels` . 

but it requires 'What is the ID?' . It is never seen before with filament package  v3. 

<img width="552" alt="Screenshot 2023-10-27 at 1 53 48 AM" src="https://github.com/filamentphp/filament/assets/69987486/ece4b3d1-8670-40d7-94b0-ea58903b8192">

What is this mean?  And what should I do?

**Comment by leandrocfe** [comment]
You should use the upgrade section
https://filamentphp.com/docs/3.x/panels/upgrade-guide/

This will make changes to your code

**Reply by magic-thomas** [reply]
The process of automatic upgrading does not work for me. 

It gives this error and can not be fixed with my understanding.

Even `Filament\PanelProvider::class` included in files and in app.php, the same error is seen.


<img width="394" alt="Screenshot 2023-10-27 at 2 31 33 AM" src="https://github.com/filamentphp/filament/assets/69987486/74addee6-e991-4ffd-be2b-09d24ca8c4c0">


That is why I remove and reinstall packages.

**Reply by leandrocfe** [reply]
I think you aren't running v3
Run `php artisan about`
What is the Filament version?

**Reply by magic-thomas** [reply]
this is the list what I did to follow the guide.

```bash
composer require filament/upgrade:"^3.0-stable" -W --dev
 
vendor/bin/filament-v3
```

`PanelProvider::class` added in app.php

```php
// in config/app.php

'providers' => ServiceProvider::defaultProviders()->merge([
        ;
        PanelProvider::class,
    ])->toArray(),
```



```bash
php artisan filament:install
```

It gives this error. 
<img width="386" alt="Screenshot 2023-10-27 at 2 46 26 AM" src="https://github.com/filamentphp/filament/assets/69987486/6db1bd2b-5335-4233-b643-bead1b8cee69">



and this is packages installed. 

<img width="766" alt="Screenshot 2023-10-27 at 2 46 43 AM" src="https://github.com/filamentphp/filament/assets/69987486/0c0fa352-576d-4e9d-b421-a219f6dc883f">

**Reply by leandrocfe** [reply]
What is the output when you run `vendor/bin/filament-v3` ?

**Reply by magic-thomas** [reply]
This is the commands and result in console with two command line. 

`composer require filament/upgrade:"^3.0-stable" -W --dev`

<img width="972" alt="Screenshot 2023-10-27 at 2 58 44 AM" src="https://github.com/filamentphp/filament/assets/69987486/8a9775ec-f3f9-4b9d-b563-e618098cf9da">


Anyway there was no error. 

And 

`vendor/bin/filament-v3`

<img width="890" alt="Screenshot 2023-10-27 at 2 59 16 AM" src="https://github.com/filamentphp/filament/assets/69987486/fd2e22d8-5a44-41e4-857c-0e761b8dfb06">


You can see the command `php artisan filament:install` at the last part of above capture.

it has been run after editing app.php

**Reply by magic-thomas** [reply]
I am adding what I tried more. 

After getting that error, I followed the comments. 

`composer require filament/filament:"^3.0-stable" -W --no-update`

<img width="560" alt="Screenshot 2023-10-27 at 3 05 27 AM" src="https://github.com/filamentphp/filament/assets/69987486/4833f435-70d5-4dcc-9e98-aff80ddf0c76">

Same error happened.

**Reply by leandrocfe** [reply]
Comment the AdminPanelProvider::class in the config/app.php

Run 
`composer require filament/filament -W`
`composer update`

**Reply by leandrocfe** [reply]
Run `php artisan about`
What is the output?

**Reply by magic-thomas** [reply]
php artisan command always gives error in this case. 


<img width="369" alt="Screenshot 2023-10-27 at 6 10 00 PM" src="https://github.com/filamentphp/filament/assets/69987486/13f06b7e-d9c3-437d-81e1-6a75827a5ac6">


`composer require filament/filament -W` gives same error.

**Reply by leandrocfe** [reply]
Check your config/app.php file. In 'providers' remove or comment this `App\Providers\Filament\AdminPanelProvider::class` or `PanelProvider::class` as you said

Run composer require and composer update again.

After, run `php artisan about` and post the output

**Reply by magic-thomas** [reply]
I removed `App\Providers\Filament\AdminPanelProvider::class` 

And I got this . 

<img width="878" alt="Screenshot 2023-10-27 at 7 14 52 PM" src="https://github.com/filamentphp/filament/assets/69987486/a99f54df-1c76-45fb-a9d8-939f28e25e27">

**Reply by leandrocfe** [reply]
Did you run
`composer require filament/filament -W --no-update`
`composer update`

Again?

**Reply by magic-thomas** [reply]
I guess no. 

After check your last comment, I tried `composer require filament/filament -W`. 

It gave me same error. 


```bash
In AdminPanelProvider.php line 22:

  Class "Filament\PanelProvider" not found

```

**Reply by leandrocfe** [reply]
Did you remove App\Providers\Filament\AdminPanelProvider::class and PanelProvider::class?

**Reply by magic-thomas** [reply]
Ok.. 

I removed 'App\Providers\Filament\AdminPanelProvider::class'. 

```bash
composer require filament/filament -W --no-update 

```

That is ok. 

```bash 
composer update
```

It gives dependency errors. 

<img width="881" alt="Screenshot 2023-10-27 at 7 48 32 PM" src="https://github.com/filamentphp/filament/assets/69987486/050988f0-a6a4-427f-b570-312054c842b7">


<img width="889" alt="Screenshot 2023-10-27 at 7 48 46 PM" src="https://github.com/filamentphp/filament/assets/69987486/a5599cca-265f-4968-ad52-b0e1e6518491">

**Reply by magic-thomas** [reply]
Oh.. 

I removed livewire/livewire.. 

I guess it is working. now.

**Reply by leandrocfe** [reply]
Nice. If it is working, you are able to run `php artisan filament:install`

Confirm the filament version running `php artisan about`

**Reply by magic-thomas** [reply]
After running filament::install 

I checked the version filament with `php artisan about`. 

it is upgraded to v 3. 


So I made User Resource 

```sh
    pa make:filament-resource User --generate --view
``````

But it gives me error in User Resource page. 

```consoel 
Property [$numberOfPaginatorsRendered] not found on component: [app.filament.resources.user-resource.pages.list-users]
```

Even with installation filament with panel flag, it did not remove the error. 

I will recode all Filament admin parts with fresh laravel project. 

I think that would be more time saving. 


I learned a lot from you.

Really thank you for help up to now.

**Reply by leandrocfe** [reply]
Run `php artisan filament:upgrade`
If you have published views, you should remove it.

**Comment by sinnbeck** [comment]
It is just the name of the panel you want it to create. New feature in latest minor version https://github.com/filamentphp/filament/pull/9287

**Reply by magic-thomas** [reply]
I don't get the idea of 'panel id'. 

I understand that something is going under admin panel parts. But I can not catch  whole picture.

Can you give me some tip?

**Reply by sinnbeck** [reply]
You can have multiple panels. One for admins. One for tenents. One for regular users.. So this is the name (identifier) of the panel you want to create

You can see it here after the panel is created
```php
        return $panel
            ->id('admin')

**Reply by magic-thomas** [reply]
Thank you for comments.. I guess I really got the idea of panel id .. 

```php 
        return $panel
            ->id('admin')
```

Actually because there is no explanations about this on documents, I was in trouble to sink in the admin panel .

Really helpful comment. 

Thanks.

**Comment by verybigelephants** [comment]
this strange sounding question is showing in default filament installation also and it's really non-explaining and documentation doesn't say what it is neither. i had to search for some random online tutorial to see what it is with some limited explanation https://tighten.com/insights/filament-crash-course-create-a-customizable-admin-panel-in-minutes/
